Mathlib/Data/Fintype/Pi.lean

```diff
@@ -109,3 +110,18 @@ theorem Finset.univ_pi_univ {α : Type*} {β : α → Type*} [DecidableEq α] [F
     (Finset.univ.pi fun a : α => (Finset.univ : Finset (β a))) = Finset.univ := by
   ext; simp
 #align finset.univ_pi_univ Finset.univ_pi_univ
+
+@[simp]
+lemma Fin.cons_mem_piFinset_iff {α} {n : ℕ} (p : (Fin n → α)) (x : α) (S : Finset α) :
```

Comment:
Can you add the `Fin.snoc` lemma too for good measure?

Reply:
Ok done, but I had to add API to do it, so the PR now touches Fin.Basic